### PR TITLE
Add support for custom naming strategies

### LINKED ISSUE #117: A way to custom case classe to swagger definition generation
We are using Scala in our API. And as per scala standards, case classes need to follow camelCasing while the response json is in snake_case. So when we provide mapping for swagger spec generation, the case class we provide to $ref for response schema again follows camelCasing.

So is there any way to provide json serialization so that the required conversions can be done???

For example: 
case class Person(id:Int, firstName: String, permanentAddress : String) 

which currently appears as below on Swagger-UI : 
{
"id" : 123,
"firstName" : "myname",
"permanentAddress" : "xyz-address"
}

while the response must look like: 
{
"id" : 123,
"first_name" : "myname",
"permanent_address" : "xyz-address"
}

### PR DESCRIPTION
**Pull request by gabfssilva**
Fixes #117 and #284.

Question: Were I supposed to work on the 2.7 branch or the master one? If I were, I can send another PR.

### CODE REVIEW

File: core/src/main/scala/com/iheart/playSwagger/DefinitionGenerator.scala [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
-import com.iheart.playSwagger.Domain.{ CustomMappings, SwaggerParameter, GenSwaggerParameter, Definition }
+import com.iheart.playSwagger.Domain.{ CustomMappings, Definition, GenSwaggerParameter, SwaggerParameter }
 import com.iheart.playSwagger.SwaggerParameterMapper.mapParam
 import play.routes.compiler.Parameter
 
 import scala.reflect.runtime.universe._
 
 final case class DefinitionGenerator(
+  namingStrategy: NamingStrategy       = NamingStrategy.None,
```
Comment (kailuowang) [inline core/src/main/scala/com/iheart/playSwagger/DefinitionGenerator.scala]: can we move this new argument to the last? That way, combined with defaut value, it's not source breaking.

Comment (gabfssilva) [reply core/src/main/scala/com/iheart/playSwagger/DefinitionGenerator.scala]: Sure. I'll push the changes right away.

**Comment by gabfssilva** [comment]
So, I thought I could do the same with `SwaggerSpecGenerator`, but, since it has lots of `apply` methods with var args parameters, it was better to leave it as the first parameter.

**Comment by kailuowang** [comment]
I tried to merge / cherry-pick this to the play2.7 branch but there are too many conflicts, mainly because of this https://github.com/iheartradio/play-swagger/pull/279 is merged to Play2.7 but not master. 
Would you be interested in submitting another PR of this feature to the play2.7 branch?

**Comment by gabfssilva** [comment]
Sure. I'll do that.

**Pull request merged**